### PR TITLE
[silgenpattern] Change catch emission to not be conditional on owners…

### PR DESCRIPTION
**Pull request by gottesmm**
…hip verification being enabled.

We were missing the code-coverage needed to show this so I added some examples
to test/SILGen/errors.swift.

rdar://29791263

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: test/SILGen/errors.swift [hunk 1]
```diff
@@ -176,6 +176,79 @@ func all_together_now(_ flag: Bool) -> Cat {
...
+// CHECK:   store [[COPIED_ERROR]] to [init] [[CAST_INPUT_MEM:%.*]] : $*Error
+// CHECK:   checked_cast_addr_br copy_on_success Error in [[CAST_INPUT_MEM]] : $*Error to HomeworkError in [[CAST_OUTPUT_MEM:%.*]] : $*HomeworkError, [[CAST_YES_BB:bb[0-9]+]], [[CAST_NO_BB:bb[0-9]+]],
+//
+// CHECK: [[CAST_YES_BB]]:
+// CHECK:   [[SUBERROR:%.*]] = load [take] [[CAST_OUTPUT_MEM]]
+// CHECK:   switch_enum [[SUBERROR]] : $HomeworkError, case #HomeworkError.TooHard!enumelt: {{bb[0-9]+}}, default [[SWITCH_MATCH_FAIL_BB:bb[0-9]+]],
+//
+// CHECK: [[SWITCH_MATCH_FAIL_BB]]([[SUBERROR:%.*]] : @owned $HomeworkError):
+// CHECK:   destroy_value [[SUBERROR]]
+// CHECK:   br [[RETHROW_BB:bb[0-9]+]]([[ERROR]] : $Error)
```
Comment (atrick) [inline test/SILGen/errors.swift]: Does there need to be an `end_borrow` in `SWITCH_MATCH_FAIL_BB1`.

File: test/SILGen/errors.swift [hunk 2]
```diff
@@ -176,6 +176,79 @@ func all_together_now(_ flag: Bool) -> Cat {
...
+//
+// CHECK: [[CAST_NO_BB]]:
+// CHECK:   br [[RETHROW_BB]]([[ERROR]] : $Error)
```
Comment (atrick) [inline test/SILGen/errors.swift]: Does there need to be an `end_borrow` in `CAST_NO_BB`.

File: test/SILGen/errors.swift [hunk 3]
```diff
@@ -176,6 +176,79 @@ func all_together_now(_ flag: Bool) -> Cat {
...
+// CHECK:   [[COPIED_ERROR:%.*]] = copy_value [[BORROWED_ERROR]]
+// CHECK:   store [[COPIED_ERROR]] to [init] [[CAST_INPUT_MEM:%.*]] : $*Error
+// CHECK:   checked_cast_addr_br copy_on_success Error in [[CAST_INPUT_MEM]] : $*Error to HomeworkError in [[CAST_OUTPUT_MEM:%.*]] : $*HomeworkError, [[CAST_YES_BB:bb[0-9]+]], [[CAST_NO_BB:bb[0-9]+]],
+//
+// CHECK: [[CAST_YES_BB]]:
+// CHECK:   [[SUBERROR:%.*]] = load [take] [[CAST_OUTPUT_MEM]]
+// CHECK:   switch_enum [[SUBERROR]] : $HomeworkError, case #HomeworkError.TooHard!enumelt: {{bb[0-9]+}}, case #HomeworkError.TooMuch!enumelt: {{bb[0-9]+}}, default [[SWITCH_MATCH_FAIL_BB:bb[0-9]+]],
+//
+// CHECK: [[SWITCH_MATCH_FAIL_BB]]([[SUBERROR:%.*]] : @owned $HomeworkError):
+// CHECK:   destroy_value [[SUBERROR]]
```
Comment (atrick) [inline test/SILGen/errors.swift]: Same question about the `end_borrow` here.

File: test/SILGen/errors.swift [hunk 4]
```diff
@@ -176,6 +176,79 @@ func all_together_now(_ flag: Bool) -> Cat {
   }
 }
 
+// Make sure that if we catch an error in a throwing function we borrow the
+// error and only consume the error by the rethrow block.
```
Comment (atrick) [inline test/SILGen/errors.swift]: only consume the error _in_ the rethrow block.

**Review by atrick — APPROVED** [review]
LGTM

**Comment by gottesmm** [comment]
@atrick made the changes. Merging!

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Pull request merged**